flooddrake/timestepper.py

```diff
@@ -38,6 +40,33 @@ def __init__(self, V, bed, source, MaxTimestep=0.025, func=lambda x: 1):
 
         self.V = V
 
+        # boundary conditions - default at solid wall for makers not given
+        self.boundary_conditions = boundary_conditions
+        if self.boundary_conditions is None:
+            if self.mesh.geometric_dimension() == 1:
+                self.boundary_conditions = [BoundaryConditions(1),
+                                            BoundaryConditions(2)]
+            if self.mesh.geometric_dimension() == 2:
+                self.boundary_conditions = [BoundaryConditions(1),
+                                            BoundaryConditions(2),
+                                            BoundaryConditions(3),
+                                            BoundaryConditions(4)]
```

Comment:
Yes indeed there is. Boundary tags are shown in `mesh.topology.exterior_facets.unique_markers` so we can loop over these